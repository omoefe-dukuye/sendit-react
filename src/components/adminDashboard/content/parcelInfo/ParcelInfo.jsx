import React from 'react';
import PropTypes from 'prop-types';
import Map from '../../../map';
import './styles.scss';

const ParcelInfo = ({ parcel }) => {
  const {
    id,
    pickup_location: from,
    destination: to,
    current_location: location,
    weight,
    distance,
    price,
    sent_on: createdOn,
    coords: [pickupLocation, initialCenter, destination]
  } = parcel;

  return (
    <div className='parcel-info--wrapper'>
      <div className='parcel-info--info--wrapper'>
        <div className='parcel-info--info'>
          <ul>
            <li>ID: {id}</li>
            <li>From: {from}</li>
            <li>To: {to}</li>
            <li>Location: {location}</li>
            <li>Weight: {weight}</li>
            <li>Total Distance: {distance}</li>
            <li>Price: {price}</li>
            <li>Created On: {createdOn}</li>
          </ul>
        </div>
      </div>
      <div className='parcel-info--map'>
        <Map initialCenter={initialCenter} pickupLocation={pickupLocation} destination={destination} />
      </div>
    </div>
  );
};

ParcelInfo.propTypes = {
  coords: PropTypes.array,
  parcel: PropTypes.object
};

export default ParcelInfo;
