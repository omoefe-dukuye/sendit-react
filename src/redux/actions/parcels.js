import { toast } from "react-toastify";
import axios from "../../utils/axiosConfig";
import { ADD_ORDER } from "./types";

const api = process.env.API_ROOT_URL;

export const createOrder = (payload) => async (dispatch) => {
  try {
    const { data: { parcel } } = await axios.post(`${api}/parcels`, payload);

    dispatch({
      type: ADD_ORDER,
      payload: parcel
    });

    return parcel;
  } catch ({ response: { data: { error } } }) {
   toast.error(error);
  }
}

export default createOrder;
