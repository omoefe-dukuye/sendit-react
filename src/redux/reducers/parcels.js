import { ADD_ORDER } from '../actions/types';

const parcelReducerDefaultState = [];

export default (state = parcelReducerDefaultState, action) => {
  switch(action.type) {
    case ADD_ORDER:
      return [...state, action.payload];

    default:
      return state;
  }
};
