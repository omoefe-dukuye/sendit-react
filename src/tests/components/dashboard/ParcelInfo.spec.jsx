import React from 'react';
import { shallow } from 'enzyme';
import '@babel/polyfill';
import ParcelInfo from '../../../components/adminDashboard/content/parcelInfo';
import parcels from '../../mocks/mockParcels';

test('Should render static component', () => {
  const wrapper = shallow(<ParcelInfo parcel={parcels[0]} />);
  expect(wrapper).toMatchSnapshot();
});
