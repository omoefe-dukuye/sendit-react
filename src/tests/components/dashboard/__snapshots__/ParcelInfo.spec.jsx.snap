// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render static component 1`] = `
<div
  className="parcel-info--wrapper"
>
  <div
    className="parcel-info--info--wrapper"
  >
    <div
      className="parcel-info--info"
    >
      <ul>
        <li>
          ID: 
          34
        </li>
        <li>
          Placed by: 
          14
        </li>
        <li>
          From: 
          1 Aminu St, Mende, Lagos, Nigeria
        </li>
        <li>
          To: 
          Mushin, Lagos, Nigeria
        </li>
        <li>
          Location: 
          1 Aminu St, Mende, Lagos, Nigeria
        </li>
        <li>
          Weight: 
          5
        </li>
        <li>
          Total Distance: 
          5
        </li>
        <li>
          Price: 
          N 625
        </li>
        <li>
          Created On: 
          March 27th 2019, 9:29:57 pm
        </li>
      </ul>
    </div>
  </div>
  <div
    className="parcel-info--map"
  >
    <Wrapper
      destination={
        Object {
          "lat": 6.601838,
          "lng": 3.3514863,
        }
      }
      initialCenter={
        Object {
          "lat": 6.570213,
          "lng": 3.3731765,
        }
      }
      pickupLocation={
        Object {
          "lat": 6.570213,
          "lng": 3.3731765,
        }
      }
    />
  </div>
</div>
`;
